conn.go

```diff
@@ -724,33 +719,28 @@ type TraceTag struct {
 	Action string
 }
 
-const userpwCtxKey = ctxKey("userPw")
+const paramsCtxKey = ctxKey("params")
 
-// ContextWithUserPassw returns a context with the specified user and password,
-// to be used with heterogeneous pools.
-func ContextWithUserPassw(ctx context.Context, user, password, connClass string) context.Context {
-	return context.WithValue(ctx, userpwCtxKey, [3]string{user, password, connClass})
+//-----------------------------------------------------------------------------
+// ContextWithParams()
+//   Returns a context with the specified parameters. These parameters are used
+// to modify the session acquired from the pool. If a standalone connection is
+// being used this will have no effect.
+//-----------------------------------------------------------------------------
+func ContextWithParams(ctx context.Context, params ConnParams) context.Context {
+    return context.WithValue(ctx, paramsCtxKey, params)
 }
 
-func (c *conn) ensureContextUser(ctx context.Context) error {
```

Comment:
Take a look at connector.Connect(). That accepts the context and the parameters are acquired from that context and used to create the connection. As such, there should be no need to "ensure" the context user as it should already be ensured. It does mean that you can't *change* the context user once you have the connection -- but I think that is appropriate. Would you agree?

Reply:
Ok, I've had to check, but it seems that if the driver implements driver.DriverContext (has an OpenConnector method), then it will be used instead of Open.

So this shoukd work - with the trivial exception when the user has a fixed sql.Conn or sql.Tx instead of an sql.DB.